internal/cache/file/cache_handle.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/data"
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/file/downloader"
+	"github.com/googlecloudplatform/gcsfuse/internal/cache/lru"
+	"github.com/googlecloudplatform/gcsfuse/internal/logger"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+)
+
+const (
+	InvalidFileHandle      = "destroy this cache_handle: invalid file handle"
+	InvalidFileDownloadJob = "destroy this cache_handle: invalid download job"
```

Comment:
Invalid means, it can be nil as well as some wrong value. So, in some cases, we might point to INVALID job in that case too we will use this message.

InvalidFileDownloadJobErrMsg ? Or let me know - if I am missing something.

Reply:
SG, in that case, InvalidFileInfoCache = "file info cache is nil" -> InvalidFileInfoCache = "invalid file info cache"